Core/source/platform/util.ts

```diff
@@ -30,3 +31,80 @@ export const setGlobals = () => {
 export const iso2022jpToUtf = (content: Buf) => {
   return dereq_encoding_japanese.convert(content, { to: 'UTF8', from: 'JIS', type: 'string' });
 }
+
+/**
+ * Create hex string from a binary
+ * @param {String} str String to convert
+ * @returns {String} String containing the hexadecimal values
+ */
+export const str_to_hex = (str: string): string => {
+  if (str === null) {
+    return "";
+  }
+  const r = [];
+  const e = str.length;
+  let c = 0;
+  let h;
+  while (c < e) {
+    h = str.charCodeAt(c++).toString(16);
+    while (h.length < 2) {
+      h = "0" + h;
+    }
+    r.push("" + h);
+  }
+  return r.join('');
+}
+
+const maxDate = (dates: (Date | null)[]): Date | null => {
+  let res: Date | null = null;
+  for (const date of dates) {
+    if (res == null || (date != null && date > res)) {
+      res = date;
+    }
+  }
+  return res;
+}
+
+const getSubkeyExpirationTime = (subkey: Subkey): number | Date => {
+  const bindingCreated = maxDate(subkey.bindingSignatures.map(b => b.created));
+  const binding = subkey.bindingSignatures.filter(b => b.created === bindingCreated)[0];
+  return binding.getExpirationTime();
+}
+
+// Attempt to backport from openpgp.js v4
+export const getKeyExpirationTimeForCapabilities = async (
```

Comment:
Thank you. I think it would be appropriate to review our code and ensure we need this functionality. I think we do, to make sure the key we encrypt for is not expired. It depends on how OpenPGP.js v5 handles this by default:
 - if their `getExpirationTime` looks at longes expiration regardless of usage, then that's not good - it will blow up when actually trying to encrypt
 - if their algo looks at intersection, that may work for us, but I think it's unlikely they designed it that way

This would be for another issue

Reply:
I haven't inspected this `getKeyExpirationTimeForCapabilities` method in detail yet